components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -1598,6 +1618,36 @@ List<OperationPolicyData> getAllAPISpecificOperationPolicies(String apiUUID, Str
     void deleteOperationPolicyById(String policyId, String organization) throws APIManagementException;
 
     /**
+     * Delete an API endpoint by providing the endpoint ID.
+     *
+     * @param endpointId     API Endpoint UUID
+     * @throws APIManagementException
+     */
+    void deleteAPIEndpointById(String endpointId) throws APIManagementException;
+
+    /**
+     *  Update an endpoint by providing the endpoint ID.
+     *
+     * @param endpointUUID Unique identifier of endpoint
+     * @param apiEndpoint  Endpoint with updated details
+     * @return
+     * @throws APIManagementException
+     */
+    APIEndpointInfo updateAPIEndpoint(String endpointUUID, APIEndpointInfo apiEndpoint)
+            throws APIManagementException;
+
+    /**
+     * Insert new endpoint for an API.
+     *
+     * @param apiUUID Unique identifier of API
+     * @param apiEndpoint New Endpoint payload object
+     * @return
+     * @throws APIManagementException
+     */
+    String addAPIEndpoint(String apiUUID, APIEndpointInfo apiEndpoint) throws APIManagementException;
```

Comment:
since `apiUUID` can be either UUID of the API or a revision Id of the API, better to document that in the method description.

Reply:
ok, I will make a proper description.

Reply:
Sorry, I misunderstood your statement. Add API endpoint always add endpoint without revision which means revision UIUID is null.
Revisions of endpoints handle by adding revision/restore revision methods which handle other part revisions of an API.

Reply:
ah okay. Can you link that method in a comment?

Reply:
https://github.com/Sharma-PsW/carbon-apimgt/blob/feature/APIEndpoint/components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java#L1649

In my branch, I added comments on what you mentioned here. I don't know why this wasn't changed into the outdated state.